meson.build

```diff
@@ -75,7 +75,11 @@ m_dep = cpp.find_library('m', required: false)
 
 freetype_dep = null_dep
 if not get_option('freetype').disabled()
-  freetype_dep = dependency('freetype2', required: false)
+  freetype_dep = dependency('freetype2', method: 'pkg-config', required: false)
```

Comment:
Can we keep the first check without specifying `method`, does not seem necessary to override the default here.

Reply:
But this whole finding FreeType business is really messed up, too much work for something that should’ve been one line. Since meson seem to have builtin support for finding specific packages with special requirements, and FreeType is such a popular package, it might be good idea to try to teach meson how to handle all of this instead of hacking every package that has a FreeType dependency.

Reply:
Hi @khaledhosny,

Yes, it would make sense, except this becomes even more complicated when we want to look for FreeType that is newer than a specific version (which we don't at this moment), meaning that Meson needs to be updated with a mapping of versions when there is a Freetype release, for instance:

From https://fossies.org/diffs/freetype/2.9.1_vs_2.10.0/docs/VERSIONS.TXT-diff.html, one will see the following:

libtool version (the one that is used in the .pc files): 22.1.16
Actual version (the one that is used in the CMake files): 2.9.1

Let me know what is the best to do here.

Reply:
Why would CMake report a different version that pkg-config, that looks like an odd choice from FreeType developers!